pommerman/envs/graphics.py

```diff
@@ -0,0 +1,175 @@
+        self._agent_tile_size =  (height / 4) / board_size
+        self._agent_count = len(agents)
+        self._agents = agents
+        self._is_partially_observable = partially_observable == True
+        self._agent_view_size = 4
+
+        glEnable(GL_BLEND)
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA)
+
+        @self.window.event
+        def close(self):
+            self.window.close()
+            self.isopen = False
+
+    def set_agent_view_size(self, view_size):
+        self._agent_view_size =view_size
+
+    def set_frame_section(self, frames):
+        self._frame_section = frames
+    
+    def clear_frame_section(self):
+        self._frame_section = []
+
+    def set_agents_state(self, agents):
+        self._agents = agents
+
+    def set_frames_from_game_state(self, state):
+        images = empty_frame(self._board_size)
+
+        for r in range(self._board_size):
+            for c in range(self._board_size):
+                item_value = state[r][c]
+                image = self._sprite_manager.get_image_from_state_value(item_value)
+                images[r][c] = image
+        
+        self.set_frame_section(images)
+
+    def render(self):
+        self.window.clear()
+        self.window.switch_to()
+        self.window.dispatch_events()
+        
+        self.render_main_board()
+        self.render_agents_board()
+        
+        self.window.flip()
+
+    def render_main_board(self):
+        sections = self._frame_section
+
+        for r in range(self._board_size):
+            for c in range(self._board_size):
+                x = c * self._tile_size
+                y = r * self._tile_size
+                sections[r][c].blit(x, y, width=self._tile_size, height=self._tile_size)
+
+    def render_agents_board(self):
+        x_offset = self._board_size * self._tile_size
+        for agent in self._agents:
+            y_offset = agent.agent_id * (self._agent_tile_size * self._board_size)
+            self.render_agent_board(agent, x_offset, y_offset)
+
+        
+
+    def render_agent_board(self, agent, x_offset, y_offset):
+        row, col = agent.position
+        agent_view_size = self._agent_view_size
+        sections = self._frame_section
+        fog = self._sprite_manager.get_fog()
+
+        for r in range(self._board_size):
+            for c in range(self._board_size):
+                x = c * self._agent_tile_size + x_offset
+                y = r * self._agent_tile_size + y_offset
+                if self._is_partially_observable and not all([
+                            row >= r - agent_view_size, row < r + agent_view_size,
+                            col >= c - agent_view_size, col < c + agent_view_size
+                    ]):
+                    fog.blit(x, y, width=self._agent_tile_size, height=self._agent_tile_size)
+                else:
+                    sections[r][c].blit(x, y, width=self._agent_tile_size, height=self._agent_tile_size)
```

Comment:
If you make `is_partially_observable` and `tile_size` into arguments (and move the first two lines inside the if), then you can actually make this a separate function (`render_board`) and call it from both `render_agent_board` and `render_main_board` (as `self.render_board(None, 0, 0, False, self._tile_size)`) and this way reuse much more code.